### PR TITLE
Revert "rz-test: Set default test dir to '.' (#4741)"

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr reverts #4741 since apparently it has code that allows tests in the official Rizin test hierarchy to find their test binaries (in `test/bins`) no matter what the current working directory is.

**Test plan**

Given the `test/db/cmd/cmd_pd` testfile pathname, all of the following should work:
1. `rz-test test/db/cmd/cmd_pd` in the root folder
2. `rz-test db/cmd/cmd_pd` in the `test/` folder
3. `rz-test cmd/cmd_pd` in the `test/db` folder
4. `rz-test cmd_pd` in the `test/db/cmd` folder

One cannot prepend `./` to the `rz-test` arguments above though (except for number 2 for some reason -- `rz-test ./db/cmd/cmd_pd` in the `test/` folder works), but that's an issue for another pr.

Also, all builds should be green.

**Closing issues**

...

### CODE REVIEW

**Comment by kazarmy** [comment]
Updated [the pr header comment](https://github.com/rizinorg/rizin/pull/4759#issue-2724444366)

File: binrz/rz-test/rz-test.c [hunk 1]
```diff
@@ -107,10 +106,95 @@ static int help(bool verbose) {
 	return 1;
 }
 
+static bool rz_test_chdir(const char *argv0) {
```
Comment (XVilka) [inline binrz/rz-test/rz-test.c:124]: Could you please add a small Doxygen comment documenting the logic?

File: binrz/rz-test/rz-test.c [hunk 2]
```diff
@@ -107,10 +106,95 @@ static int help(bool verbose) {
...
+#else
+	return false;
+#endif
+}
+
 static bool rz_test_test_run_unit(void) {
 	return rz_sys_system("make -C unit all run") == 0;
 }
 
+static bool rz_test_chdir_fromtest(const char *test_path) {
```
Comment (XVilka) [inline binrz/rz-test/rz-test.c:171]: Same

**Review by XVilka — CHANGES_REQUESTED** [review]

File: binrz/rz-test/rz-test.c (hunk 1 above)
Comment (kazarmy) [reply binrz/rz-test/rz-test.c:124]: Since I'm not a Large Language Model, this unfortunately will take some time

Comment (kazarmy) [reply binrz/rz-test/rz-test.c:124]: > Could you please add a small Doxygen comment documenting the logic?

Ok a920c5ea31596946f72f9d9e8c3f0e173c62b051

File: binrz/rz-test/rz-test.c (hunk 2 above)
Comment (kazarmy) [reply binrz/rz-test/rz-test.c:171]: Ok comment done 4dd3b6fde12bd6cd9522eaa883a6cf8a8a83476d

**Review by XVilka — APPROVED** [review]
Thank you for writing the documentation!

**Pull request merged**